Source/WTF/wtf/HexNumber.cpp

```diff
@@ -46,13 +49,37 @@ std::pair<LChar*, unsigned> appendHex(LChar* buffer, unsigned bufferSize, std::u
     return { start, end - start };
 }
 
+WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+
 }
 
 void printInternal(PrintStream& out, HexNumberBuffer buffer)
 {
     out.print(StringView(buffer.span()));
 }
 
-} // namespace WTF
+static void toHexInternal(std::span<const uint8_t> values, std::span<LChar> hexadecimalOutput)
+{
+    for (auto [i, digestValue] : IndexedRange(values)) {
+        hexadecimalOutput[i * 2] = upperNibbleToASCIIHexDigit(digestValue);
+        hexadecimalOutput[i * 2 + 1] = lowerNibbleToASCIIHexDigit(digestValue);
+    }
+}
 
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+CString toHexCString(std::span<const uint8_t> values)
+{
+    std::span<char> buffer;
+    auto result = CString::newUninitialized(values.size() * 2, buffer);
```

Comment:
Just occurred to me that this could overflow in theory even though it’s really not a practical concern. Should it perhaps use checked arithmetic or have a `RELEASE_ASSERT`?